packages/mobile/src/send/ValidateRecipientAccount.tsx

```diff
@@ -88,11 +95,22 @@ export class ValidateRecipientAccount extends React.Component<Props, State> {
     isModalVisible: false,
   }
 
-  componentDidUpdate = (prevProps: Props) => {
-    const { isValidRecipient, isOutgoingPaymentRequest, transactionData } = this.props
-    if (isValidRecipient && !prevProps.isValidRecipient) {
+  componentDidMount = () => {
+    const { e164PhoneNumber } = this.props.recipient
+    if (e164PhoneNumber) {
+      this.props.validateRecipientAddressReset(e164PhoneNumber)
+    }
+  }
+
+  componentDidUpdate = () => {
+    const { validationSuccessful, isOutgoingPaymentRequest, transactionData } = this.props
+
+    if (validationSuccessful) {
```

Comment:
We should check `validationSuccessful` actually changed, otherwise we risk triggering this path unnecessarily as the code evolves and other updates are triggered.

Reply:
You're right, better to be safe on this.